CMSgov/managed-care-review
# fix: MR-2542 Ensure that trailing spaces in email address inputs do not trigger errors

## Summary
Looking for feedback on the approach. I tried addressing the bug in two ways
- Using `.trim` in yup validations. This ensures validation error is simply not raised. See 482e326ea69c9bc0f161dbfe9ab047410aca1fe5 
- Direct remove leading and trailing whitespace from the input value itself `onBlur` to prevent this class of issue from arising, at least where our app-wide components are used . See 6ccda52dc6f5825503d1c6d447da43b887eaad4d 

Other change in this PR is slimming contacts and actuary contacts JSX by using `<FieldTextInput />` to remove extraneous markdown. 

#### Related issues
https://qmacbis.atlassian.net/browse/MR-2542

#### Test cases covered


## QA guidance
- Besides testing bugfix to email address inputs, worth running through contacts and actuary contacts UI, make sure I didn't cause a regression here. Seems like we are testing with unit tests and Cypress pretty thoroughly here which is good to see.
